ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/JdbcLedgerDao.scala

```diff
@@ -590,13 +590,24 @@ private class JdbcLedgerDao(
   }
 
   override def prune(
-      pruneUpToInclusive: Offset
+      pruneUpToInclusive: Offset,
+      pruneAllDivulgedContracts: Boolean,
   )(implicit loggingContext: LoggingContext): Future[Unit] =
     dbDispatcher.executeSql(metrics.daml.index.db.pruneDbMetrics) { conn =>
-      storageBackend.pruneEvents(pruneUpToInclusive)(conn)
+      storageBackend.pruneEvents(pruneUpToInclusive, pruneAllDivulgedContracts)(conn)
       storageBackend.pruneCompletions(pruneUpToInclusive)(conn)
       storageBackend.updatePrunedUptoInclusive(pruneUpToInclusive)(conn)
+
+      if (pruneAllDivulgedContracts) {
+        storageBackend.updatePrunedAllDivulgenceEventsUpToInclusive(pruneUpToInclusive)(conn)
+      }
+
       logger.info(s"Pruned ledger api server index db up to ${pruneUpToInclusive.toHexString}")
+
+      if (pruneAllDivulgedContracts)
+        logger.info(
+          s"Pruned all divulgence events in the ledger api server index db up to ${pruneUpToInclusive.toHexString}"
+        )
```

Comment:
Consider making it one log-message: "Pruned ledger api server index db (including all divulgence events) up to ... "

Also: I believe the log message is misleading, as it is output before the transaction commit completed. Given that it is a long-running change I'd suggest the following:
1. emit a log message stating "Starting pruning of ledger api server index db .. " before the first `storageBackend` call
2. emit a log message stating "Completed pruning ... <same message as above>" outside the `dbDispatcher` call, if it was successful, and an error with "Failed to prune ... due to <exception>" if an exception was raised.

Reply:
Thanks, will address the log message. Wrt to the transaction failure, I've moved the logging at the end of it for that specific reason.

Reply:
You're suggestion is better though with logging outside the dispatcher call.